Data/Map/Strict.hs

```diff
@@ -15,19 +15,63 @@
 -- Maintainer  :  libraries@haskell.org
 -- Portability :  portable
 --
--- An efficient implementation of ordered maps from keys to values
--- (dictionaries).
 --
--- API of this module is strict in both the keys and the values.
--- If you need value-lazy maps, use "Data.Map.Lazy" instead.
--- The 'Map' type is shared between the lazy and strict modules,
--- meaning that the same 'Map' value can be passed to functions in
--- both modules (although that is rarely needed).
+-- = Finite Maps (strict interface)
 --
--- These modules are intended to be imported qualified, to avoid name
--- clashes with Prelude functions, e.g.
+-- The @'Map' k v@ type represents a finite map (sometimes called a dictionary)
+-- from keys of type @k@ to values of type @v@. A 'Map' is strict in its keys and
+-- values.
```

Comment:
No. This paragraph is just wrong. A `Map` is *lazy* in its values. Please don't say that it's strict in its values, although it's tempting to imagine that when importing only the `.Strict` module. The part below should say something like

```haskell
-- Each function in this module is careful to force values before installing them in a 'Map'. This
-- is usually more efficient when laziness is not necessary. When laziness *is* required, use
-- the functions in "Data.Map.Lazy".
--
-- In particular, the functions in this module obey the following law:
--
--     If all values stored in all maps in the arguments are in WHNF, then all values stored
--     in all maps in the results will be in WHNF once those maps are evaluated.
--
-- Examples:
--
-- @
-- 'insert' k undefined m == undefined
-- 'insert' 1 'a' $ "Data.Map.Lazy".'Data.Map.Lazy.fromList' ((2,undefined) : zip [3 .. 9] ['b'..])
--   == "Data.Map.Lazy".'Data.Map.Lazy.fromList' ((1,'a') : (2, undefined) : zip [3..9] ['b'..])
-- 'union' ('fromList' $ zip [1..10] ['a'..]) ("Data.Map.Lazy".'Data.Map.Lazy.fromList' $ zip [1..10] (repeat undefined)) == 'fromList' $ zip [1..10] ['a'..])
-- 'unionWith' (\_ _ -> undefined) m1 m2 == undefined
-- 'alterF' (\_ -> pure (Just undefined)) k m == pure undefined
-- @
```